### PR TITLE
Supporting User MSI and CMK in RS Vault CRUD Operation

### PR DESCRIPTION
**Pull request by asmaskar**
…D (#1)

* Supporting User Assigned Identity and CMK.

* Indentation fixes

* Added Examples for CRUD with CMK

* Added more fields to the Operation Resource

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [x] PowerShell
      - [x] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json [hunk 1]
```diff
@@ -452,19 +459,29 @@
               "$ref": "#/definitions/Vault"
             }
           },
-          "201": {
-            "description": "Created",
-            "schema": {
-              "$ref": "#/definitions/Vault"
-            }
+          "202": {
```
Comment (jorgecotillo) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json:462]: What's the reason behind updating the response code?

Comment (asmaskar) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json:462]: The reason is that 201 was not being used from the service side. The next reason is that the synchronous call has been made Asynchronous so 202 is added as per the ARM guidelines

Comment (jorgecotillo) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json:462]: Good, I realized it after looking at the new operationStatus endpoints but it wasn't clear from the PR description nor from the title.

**Comment by jorgecotillo** [comment]
From ARM side, this PR looks ok but contains new properties to an existing api-version. Please follow the breaking change process and get an approval.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json (hunk 1 above)
Comment (asmaskar) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json:462]: Sorry for the confusion.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json [hunk 2]
```diff
@@ -606,6 +623,99 @@
...
+            "name": "operationId",
+            "in": "path",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/ApiVersion"
+          }
+        ],
+        "responses": {
```
Comment (filizt) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json:657]: Missing error response model. Consider using error models defined in types.json in common-types folder: [https://github.com/Azure/azure-rest-api-specs/tree/master/specification/common-types/resource-management](https://github.com/Azure/azure-rest-api-specs/tree/master/specification/common-types/resource-management)

**Comment by filizt** [comment]
>         "responses": {

Missing error response model here and in other updated paths. 

---
Refers to: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json:455 in 4fe5b9a. [](commit_id = 4fe5b9a5ea6a3e1e9eb7e91e85c5207323f8fda8, deletion_comment = False)

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json (hunk 2 above)
Comment (majastrz) [reply specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2016-06-01/vaults.json:657]: Yeah, this is actually an issue on a lot of the responses in this swagger.

---
In reply to: [551566447](https://github.com/Azure/azure-rest-api-specs/pull/12248#discussion_r551566447) [](ancestors = 551566447)

**Comment by filizt** [comment]
Added a few comments - please update the PR when ready.

**Comment by asmaskar** [comment]
Hi,
In this iteration, I have incorporated the feedback by the reviewers and also made the changes in a new api-version.
I have kept the older version without any changes.
Please let me know in case of any issues

**Comment by asmaskar** [comment]
@filizt Can you please review this and let me know if there are any concerns around the PR

**Comment by filizt** [comment]
> @filizt Can you please review this and let me know if there are any concerns around the PR

I'll take a look. Could you also mark all the checkboxes that apply to your changes on the PR?

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2020-02-02/vaults.json [hunk 3]
```diff
@@ -0,0 +1,1657 @@
...
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/Vault"
+            }
+          },
+          "202": {
+            "description": "Accepted"
+          },
```
Comment (filizt) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2020-02-02/vaults.json:518]: Please mark as long running operation.

File: specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2020-02-02/vaults.json [hunk 4]
```diff
@@ -0,0 +1,1657 @@
...
```
Comment (filizt) [inline specification/recoveryservices/resource-manager/Microsoft.RecoveryServices/stable/2020-02-02/vaults.json:795]: Please mark as long running operation

**Comment by filizt** [comment]
@asmaskar there is only one comment waiting (https://github.com/Azure/azure-rest-api-specs/pull/12248#discussion_r554084426). If it's addressed, we should be able to approve today.

**Comment by asmaskar** [comment]
@filizt I have addressed the comment provided. Thanks

**Comment by asmaskar** [comment]
@xccc-msft can you please let me know if there are any more changes to be made in the pr. If it seems fine, can you please approve it

**Review by xseeseesee — APPROVED** [review]

**Comment by xccc-msft** [comment]
@asmaskar Please let me know when it's ready to merge. Thanks.

**Comment by asmaskar** [comment]
@xccc-msft please go ahead and merge it. Thanks

**Pull request merged**